roomserver/storage/postgres/storage.go

```diff
@@ -68,8 +71,18 @@ func (d *Database) StoreEvent(
 		}
 	}
 
-	// TODO: Room version here
-	if roomNID, err = d.assignRoomNID(ctx, nil, event.RoomID(), "1"); err != nil {
+	// Get the default room version. If the client doesn't supply a room_version
+	// then we will use our configured default to create the room.
+	// https://matrix.org/docs/spec/client_server/r0.6.0#post-matrix-client-r0-createroom
+	// Note that the below logic depends on the m.room.create event being the
+	// first event that is persisted to the database when creating or joining a
+	// room.
```

Comment:
Can we even make that guarantee over federation? I don't think we can.

Reply:
I think we would have to - without the `m.room.create` event, we can't auth anything in the newly joined room, so it really should be the first event that is sent to us. If we request those initial events somewhere then we can at least make the order of persistence determinate.

Reply:
Looking at this, `func (r RespSendJoin) Check()` in `clientapi/routing/joinroom.go` currently persists the events in the order that they appear in the `send_join` response, which is the same as a regular state response and would therefore include the `m.room.create` event.

With a small update to this function, we can look for and persist the `m.room.create` event first in all cases.

Reply:
Okay, we'll see how badly things explode.